Subversion checkout does not work for repositiories that require authentication

My scanner cannot checkout from a private SVN repo although I'm using the latest ORT version and I have set the `ORT_HTTP_USERNAME` and `ORT_HTTP_PASSWORD` environment variables.

I have added an extra info log at the beginning of the `OrtAuthenticator#getPasswordAuthentication` but I do not see it in the logs so I assume the method is not called a all. I do see the "Authenticator is already installed" message.

The auth error is:
```
"Caused by: SVNAuthenticationException: svn: E170001: Authentication required for '<http://my-repo.domain.com> Subversion Repository'".
```

Note that I have tried using a .netrc file as well but with no luck.

The `OrtSVNAuthenticationManager` class (see `Subversion.kt`) sets the `username` and `password` properties to `null` and does not seem to care about the environment variables or the `.netrc` file.

When I set username and password to the desired values in the `OrtSVNAuthenticationManager` class, the SVN checkout would work as expected.

--- a/downloader/src/main/kotlin/vcs/Subversion.kt
+++ b/downloader/src/main/kotlin/vcs/Subversion.kt
@@ -21,8 +21,10 @@
 package org.ossreviewtoolkit.downloader.vcs
 
 import java.io.File
+import java.net.Authenticator
 import java.net.InetSocketAddress
 import java.net.URI
+import java.net.URL
 import java.nio.file.Paths
 
 import org.ossreviewtoolkit.downloader.DownloadException
@@ -30,6 +32,7 @@ import org.ossreviewtoolkit.downloader.VersionControlSystem
 import org.ossreviewtoolkit.downloader.WorkingTree
 import org.ossreviewtoolkit.model.VcsInfo
 import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.utils.OrtAuthenticator
 import org.ossreviewtoolkit.utils.collectMessagesAsString
 import org.ossreviewtoolkit.utils.installAuthenticatorAndProxySelector
 import org.ossreviewtoolkit.utils.log
@@ -40,7 +43,10 @@ import org.tmatesoft.svn.core.SVNErrorMessage
 import org.tmatesoft.svn.core.SVNException
 import org.tmatesoft.svn.core.SVNNodeKind
 import org.tmatesoft.svn.core.SVNURL
+import org.tmatesoft.svn.core.auth.ISVNAuthenticationProvider
 import org.tmatesoft.svn.core.auth.ISVNProxyManager
+import org.tmatesoft.svn.core.auth.SVNAuthentication
+import org.tmatesoft.svn.core.auth.SVNPasswordAuthentication
 import org.tmatesoft.svn.core.internal.wc.DefaultSVNAuthenticationManager
 import org.tmatesoft.svn.core.io.SVNRepositoryFactory
 import org.tmatesoft.svn.core.wc.SVNClientManager
@@ -245,6 +251,37 @@ private class OrtSVNAuthenticationManager : DefaultSVNAuthenticationManager(
     /* passphrase = */ charArrayOf()
 ) {
     private val ortProxySelector = installAuthenticatorAndProxySelector()
+    private val ortAuthenticator = OrtAuthenticator.getDefaultAuthenticator()
+
+    init {
+        authenticationProvider = object : ISVNAuthenticationProvider {
+            override fun requestClientAuthentication(
+                kind: String,
+                svnurl: SVNURL,
+                realm: String,
+                errorMessage: SVNErrorMessage?,
+                previousAuth: SVNAuthentication?,
+                authMayBeStored: Boolean
+            ): SVNAuthentication? {
+                val url = URL(svnurl.toDecodedString())
+
+                val auth = ortAuthenticator?.requestPasswordAuthenticationInstance(
+                    svnurl.host, null, svnurl.port, svnurl.protocol, null, null, url, Authenticator.RequestorType.SERVER
+                ) ?: return null
+
+                return SVNPasswordAuthentication.newInstance(
+                    auth.userName, auth.password, authMayBeStored, svnurl, /* isPartial = */ false
+                )
+            }
+
+            override fun acceptServerAuthentication(
+                url: SVNURL,
+                realm: String,
+                certificate: Any,
+                resultMayBeStored: Boolean
+            ) = ISVNAuthenticationProvider.ACCEPTED
+        }
+    }
 
     override fun getProxyManager(url: SVNURL): ISVNProxyManager? {
         val proxy = ortProxySelector.select(URI(url.toDecodedString())).firstOrNull() ?: return null

--- a/utils/src/main/kotlin/OrtAuthenticator.kt
+++ b/utils/src/main/kotlin/OrtAuthenticator.kt
@@ -33,7 +33,7 @@ import org.apache.logging.log4j.Level
  */
 class OrtAuthenticator(private val original: Authenticator? = null) : Authenticator() {
     companion object {
-        private fun getDefaultAuthenticator(): Authenticator? {
+        fun getDefaultAuthenticator(): Authenticator? {
             // The getDefault() method is only available as of Java 9, see
             // https://docs.oracle.com/javase/9/docs/api/java/net/Authenticator.html#getDefault--
             val method = runCatching { Authenticator::class.java.getMethod("getDefault") }.getOrNull()

